src/System.Text.Json/src/System/Text/Json/Node/JsonString.cs

```diff
@@ -235,43 +239,50 @@ internal bool TryGetBytesFromBase64(out byte[] value)
             int bufferSize = _value.Length / 4 * 3;
 
             byte[] arrayToReturnToPool = null;
-            try
-            {
-                Span<byte> buffer = bufferSize <= JsonConstants.StackallocThreshold
-                    ? stackalloc byte[bufferSize]
-                    : arrayToReturnToPool = ArrayPool<byte>.Shared.Rent(bufferSize);
+            Span<byte> buffer = bufferSize <= JsonConstants.StackallocThreshold
+                ? stackalloc byte[bufferSize]
+                : arrayToReturnToPool = ArrayPool<byte>.Shared.Rent(bufferSize);
 
-                if (Convert.TryFromBase64String(_value, buffer, out int bytesWritten))
-                {
-                    value = buffer.Slice(0, bytesWritten).ToArray();
-                    return true;
-                }
-                else
-                {
-                    value = default;
-                    return false;
-                }
+            if (Convert.TryFromBase64String(_value, buffer, out int bytesWritten))
+            {
+                buffer = buffer.Slice(0, bytesWritten);
+                value = buffer.ToArray();
+                buffer.Clear();
+                ReturnToPool(arrayToReturnToPool);
+                return true;
             }
-            finally
+            else
```

Comment:
Hm, that was the nice part of `finally` here :wink:
With `finally` `ReturnToPool` isn't needed and with finally-cloning (by the JIT) there shouldn't be any perf-impact. So with this change there's quite some duplication and the need for a extra method.
I like the version with finally more...but with the other comment the call-sites can be simplified, so I'm ok with going finally-less.

Reply:
Hmm... good point. I am fine with either since it is generally cosmetic (we know that `Convert.TryFromBase64String` won't throw here).

Like you said, we could avoid the additional method call with the try-finally. Up to you, @AntonLandor - whatever you think is more readable.
**Try-Finally option:**
```C#
internal bool TryGetBytesFromBase64(out byte[] value)
{
    Debug.Assert(_value != null);

    // Shortest length of a base-64 string is 4 characters.
    if (_value.Length < 4)
    {
        value = default;
        return false;
    }

    // we decode string -> byte, so the resulting length will
    // be /4 * 3 - padding. To be on the safe side, keep padding and slice later
    int bufferSize = _value.Length / 4 * 3;

    byte[] arrayToReturnToPool = null;
    Span<byte> buffer = bufferSize <= 256
        ? stackalloc byte[bufferSize]
        : arrayToReturnToPool = ArrayPool<byte>.Shared.Rent(bufferSize);
    try
    {
        if (Convert.TryFromBase64String(_value, buffer, out int bytesWritten))
        {
            buffer = buffer.Slice(0, bytesWritten);
            value = buffer.ToArray();
            return true;
        }
        else
        {
            value = default;
            return false;
        }
    }
    finally
    {
        if (arrayToReturnToPool != null)
        {
            buffer.Clear();
            ArrayPool<byte>.Shared.Return(arrayToReturnToPool);
        }
    }
}
```

Method call option:
```C#
internal bool TryGetBytesFromBase64(out byte[] value)
{
    Debug.Assert(_value != null);

    // Shortest length of a base-64 string is 4 characters.
    if (_value.Length < 4)
    {
        value = default;
        return false;
    }

    // we decode string -> byte, so the resulting length will
    // be /4 * 3 - padding. To be on the safe side, keep padding and slice later
    int bufferSize = _value.Length / 4 * 3;

    byte[] arrayToReturnToPool = null;
    Span<byte> buffer = bufferSize <= 256
        ? stackalloc byte[bufferSize]
        : arrayToReturnToPool = ArrayPool<byte>.Shared.Rent(bufferSize);

    if (Convert.TryFromBase64String(_value, buffer, out int bytesWritten))
    {
        buffer = buffer.Slice(0, bytesWritten);
        value = buffer.ToArray();
        ReturnToPool(arrayToReturnToPool, buffer);
        return true;
    }
    else
    {
        value = default;
        ReturnToPool(arrayToReturnToPool, buffer);
        return false;
    }
}

private void ReturnToPool(byte[] arrayToReturnToPool, Span<byte> buffer)
{
    if (arrayToReturnToPool != null)
    {
        buffer.Clear();
        ArrayPool<byte>.Shared.Return(arrayToReturnToPool);
    }
}
```

Reply:
If there is no perf-impact with the `try-finally` I think I'd prefer it for readability, but I'd guess it's subjective. I'll most likely commit one of these solutions in a while when I've made up my mind.

Reply:
I settled for the `try-finally solution`. I personally think it's a more elegant solution.